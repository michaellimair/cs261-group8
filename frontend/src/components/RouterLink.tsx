import { Link as ReactRouterLink, LinkProps as ReactRouterLinkProps } from 'react-router-dom';
import { Link, LinkProps } from '@chakra-ui/react';
import { FC } from 'react';

/** Custom Link component which combines the `Link` component
 * from Chakra UI and the `Link` component from `react-router-dom`. */
const RouterLink: FC<LinkProps & ReactRouterLinkProps> = (props) => (
  <Link as={ReactRouterLink} {...props} />
);

export default RouterLink;
