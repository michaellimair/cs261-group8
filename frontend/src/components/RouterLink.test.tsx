import { BrowserRouter } from 'react-router-dom';
import RouterLink from './RouterLink';

describe('components/RouterLink', () => {
  it('renders properly', () => {
    const result = (
      <BrowserRouter>
        <RouterLink to="/" />
      </BrowserRouter>
    );

    expect(result).toMatchSnapshot();
  });
});
