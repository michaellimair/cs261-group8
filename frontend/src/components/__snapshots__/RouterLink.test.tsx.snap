// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/RouterLink renders properly 1`] = `
<BrowserRouter>
  <RouterLink
    to="/"
  />
</BrowserRouter>
`;
